java-frontend/src/test/java/org/sonar/java/se/Expectations.java

```diff
@@ -106,43 +118,15 @@
     }
   }
 
-  abstract static class LineRef {
-    abstract int getLine(int ref);
-
-    static LineRef fromString(String input) {
-      if (input.startsWith("+")) {
-        return new RelativeLineRef(Integer.valueOf(input));
-      } else {
-        return new AbsoluteLineRef(Integer.valueOf(input));
-      }
+  interface Issue extends Map<IssueAttribute, Object> {
+    static Issue create() {
+      return new IssueImpl();
     }
+  }
 
-    static int toLine(Object ref) {
-      return ((LineRef) ref).getLine(0);
-    }
-
-    static class AbsoluteLineRef extends LineRef {
-      final int line;
-
-      public AbsoluteLineRef(int line) {
-        this.line = line;
-      }
-
-      public int getLine(int ref) {
-        return line;
-      }
-    }
-
-    static class RelativeLineRef extends LineRef {
-      final int offset;
-
-      RelativeLineRef(int offset) {
-        this.offset = offset;
-      }
-
-      @Override int getLine(int ref) {
-        return ref + offset;
-      }
+  private static class IssueImpl extends EnumMap<IssueAttribute, Object> implements Issue {
```

Comment:
Not sure to see the point of having Issue and IssueImpl here... Why Issue can't extends EnumMap and have the create static method ? So no need for IssueImpl at all.